provider-directory/src/main/java/gov/va/api/health/providerdirectory/service/client/RestVlerClient.java

```diff
@@ -0,0 +1,171 @@
+      }
+  }*/
+
+  @SneakyThrows
+  private static <T extends VlerResponse> T handleVlerExceptions(
+      String message, Callable<T> callable) {
+    T response;
+    try {
+      response = callable.call();
+    } catch (HttpClientErrorException.NotFound e) {
+      throw new NotFound(message, e);
+    } catch (HttpClientErrorException.BadRequest e) {
+      throw new BadRequest(message, e);
+    } catch (HttpStatusCodeException e) {
+      throw new SearchFailed(message, e);
+    } catch (Exception e) {
+      throw new VlerException(message, e);
+    }
+    if (response == null) {
+      throw new VlerException(message + ", no VLER response");
+    }
+    if (response.error() != null && isNotBlank(response.error().message())) {
+      throw new VlerException(message + ", " + response.error().message());
+    }
+    return response;
+  }
+
+  @SneakyThrows
+  private String authHeader(String baseUrl, String url, String dateString) {
+    String endpoint = "/" + url.substring(url.indexOf(baseUrl) + baseUrl.length());
+    String reqStr = "GET\n" + dateString + "\napplication/json\n" + endpoint;
+    Mac encoding = Mac.getInstance("HmacSHA256");
+    encoding.init(new SecretKeySpec(prvKey.getBytes(Charset.forName("UTF-8")), "HmacSHA256"));
+    byte[] sha = encoding.doFinal(reqStr.getBytes(Charset.forName("UTF-8")));
+    String encSha = Base64.getEncoder().encodeToString(sha);
+    return "DAAS " + pubKey + ":" + encSha;
+  }
+
+  /** Calls the VLER Direct API by email address search. */
+  @Override
+  public AddressResponse endpointByAddress(String address) {
+    return handleVlerExceptions(
+        address,
+        () -> {
+          String url =
+              UriComponentsBuilder.fromHttpUrl(baseUrl + "/direct/addresses")
+                  .queryParam("cn", address)
+                  .build()
+                  .toUriString();
+          String dateString =
+              DateTimeFormatter.ofPattern("MM/dd/yyyy HH:mm:ss z").format(ZonedDateTime.now());
+          HttpHeaders headers = new HttpHeaders();
+          headers.setAccept(Collections.singletonList(MediaType.APPLICATION_JSON));
+          headers.setContentType(MediaType.APPLICATION_JSON);
+          headers.add("Authorization", authHeader(baseUrl, url, dateString));
+          headers.add("Date", dateString);
+          HttpEntity<?> requestEntity = new HttpEntity<>(headers);
+          System.out.println("RequestEntity: " + requestEntity);
+          ResponseEntity<AddressResponse> entity =
+              restTemplate().exchange(url, HttpMethod.GET, requestEntity, AddressResponse.class);
+          return entity.getBody();
+        });
+  }
+
+  /** Acceptance of SSL for Java 12. */
+  @SneakyThrows
+  public RestTemplate restTemplate() {
+    TrustStrategy acceptingTrustStrategy = (X509Certificate[] chain, String authType) -> true;
+    SSLContext sslContext =
+        org.apache.http.ssl.SSLContexts.custom()
+            .loadTrustMaterial(null, acceptingTrustStrategy)
```

Comment:
Documentation is sketchy in `SSLContexts`, but looking at the code `.loadTrustMaterial(null, acceptingTrustStrategy)` and `.loadTrustMaterial(acceptingTrustStrategy)` do the same.

Docs do say this about `TrustStrategy`
> A strategy to establish trustworthiness of certificates without consulting the trust manager configured in the actual SSL context. This interface can be used to override the standard JSSE certificate verification process.

The `(...) -> true` function, combined with not actually loading a trust store effectively disables all SSL checks on this end of the mutual TLS side. Is that what you intended?